app/controllers/auth_controller.rb

```diff
@@ -1,9 +1,35 @@
 # frozen_string_literal: true
 
 class AuthController < ApplicationController
+  THIRD_PARTY_AUTH_METHODS = {
+    okta: Okta,
+    facebook: Facebook,
+    google: GoogleAuth, # the google gem uses Google namespace
+    apple: Apple
+  }
+
   def create
-    token = data_attrs[:okta_access_token] ? auth_with_okta : auth_with_code
+    method = THIRD_PARTY_AUTH_METHODS.keys.detect { |method| data_attrs[:"#{method}_access_token"] }
+    token = case method
+    when :apple
+      # special case for apple, which has given and family name passed in
+      user = Apple.find_user_by_access_token(data_attrs[:apple_access_token], data_attrs[:apple_given_name], data_attrs[:apple_family_name])
```

Comment:
according to @levieggertcru , we are passing an `id_token` for apple auth, so the parameter should be named `apple_id_token` instead of `apple_access_token`